CorsixTH/Lua/hospitals/player_hospital.lua

```diff
@@ -386,5 +425,16 @@ function PlayerHospital:afterLoad(old, new)
     self.win_declined = false -- Has not yet declined the level win fax
   end
 
+  updateInitialsCache()
+  -- Fix the initial of all employed and employable staff
+  for _, staff in ipairs(self.staff) do
+    localiseInitial(staff.profile)
+  end
+  for _, subtable in pairs(self.world.available_staff) do
+    for _, staff in pairs(subtable) do
+      localiseInitial(staff)
+    end
+  end
```

Comment:
In multi-player, we will have several players, each possibly with a different language.
As such, I agree `PlayerHospital` is perhaps the right spot for the hospital staff.

The trouble is with `self.world.available_staff`. That will be shared between players. If three players each have a different language, what should be the name of the staff?
Difficult question.

As it is now, a staff member has as many names as there are languages, I think.

Reply:
Keeping it in `world.lua` would also work, it's a central point for all hospitals, and it does global things.

Reply:
For now: moved to world.lua

For multiplayer: Use the name seed for both parts of the name? As part of multiplayer the seed is generated once then each user makes the local name.

Reply:
That could work.

As persistence saves all lua data, it also saves your customization to the strings. If 2 players each have a different language and they both save the game, their files will be different. Don't know if it's a problem, but it's a little weird.

Perhaps it's just another aspect of multi-player I didn't know about until now.

Reply:
I don't understand why we're calling updateInitialCache and managing the afterLoad here now that the implementation is all in world. I think both the initial call and the afterLoad handling should be moved there too.

Reply:
The afterload from PlayerHospital happens before World's, and World is too late.

Reply:
That sounds like a different issue. Logically the world should be loading in before the hospitals.

Reply:
@tobylane that's not true, the afterLoad sequence is:
map (which doesn't call anything else)
ui (which calls afterLoad on all the child windows)
world (which calls afterLoad on the hospitals and other objects)

So you can control whether your update happens before or after hospitals and staff in world's afterLoad by where in the function you put your update.

Reply:
Made the move to world.

Reply:
Surprised you put it at the end of World's afterLoad, after saying that was too late. It seems to play test well for me though. What was the dependency concern?

Reply:
I must have tested wrong. The issue was writing a new initials cache before it was used, it's now in the right order.